sql/logictest: TestLogic failed

[(sql/logictest).TestLogic failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2131775&tab=buildLog) on [master@2a5eec95a7d495db1e97bbe8c330e808d9232090](https://github.com/cockroachdb/cockroach/commits/2a5eec95a7d495db1e97bbe8c330e808d9232090):

```
=== RUN   TestLogic
    TestLogic: logic.go:2761: randomize batchSize to 3
    TestLogic: test_log_scope.go:85: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestLogic885446227
    TestLogic: test_log_scope.go:58: use -show-logs to present logs inline
--- FAIL: TestLogic (997.32s)
=== RUN   TestLogic/local/txn_stats
=== PAUSE TestLogic/local/txn_stats
=== CONT  TestLogic/local/txn_stats
        --- FAIL: TestLogic/local/txn_stats (1.89s)
=== RUN   TestLogic/local
    --- FAIL: TestLogic/local (0.37s)

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestLogic PKG=./pkg/sql/logictest TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestLogic.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/sql/logictest/testdata/logic_test/txn_stats
+++ b/pkg/sql/logictest/testdata/logic_test/txn_stats
@@ -40,14 +40,16 @@ SELECT txn_count - implicit_count = 1
 true
 
 # All transactions so far should be extremely fast, so we check that the
-# average transaction time is in [0s, 0.1s] range.
+# average transaction time is in [0s, 1s] range.
+# Note: if this query flakes again, you should remove it - it's a sanity check
+# on txn_time_avg_sec field.
 query B
 SELECT count(*) = 0
   FROM crdb_internal.node_txn_stats
  WHERE application_name = 'test'
    AND (
         txn_time_avg_sec < 0
-        OR txn_time_avg_sec > 0.1
+        OR txn_time_avg_sec > 1
        )
 ----
 true

